pkg/option/resolver/resolver.go

```diff
@@ -230,12 +232,48 @@ func readConfigMap(ctx context.Context, client client.Clientset, source ConfigSo
 	return cm.Data, []string{source.String()}, nil
 }
 
-// ReadConfigOverrides reads all the CiliumNodeConfig objects and returns a flattened map
+// readNodeConfigsAllVersions read node configurations for versions v2 and v2alpha1 of CiliumNodeConfig CRD.
+// TODO depreciate CNC on v2alpha1 https://github.com/cilium/cilium/issues/31982
+func readNodeConfigsAllVersions(ctx context.Context, client client.Clientset, nodeName, namespace, name string) (map[string]string, []string, error) {
+	var errv2, errv2alpha1 error
+
+	nodeConfigv2, descv2, errv2 := readNodeConfigs(ctx, client, nodeName, namespace, name)
```

Comment:
So, the apiserver is kind of magical, and it will down-convert a v2 CNC to a v2alpha1 CNC. So, if you like, you can just retrieve the v2alpha1 version in v1.16, and switch to v2 in v1.17. Then transition v2alpha1 to non-served in v1.18

Reply:
UTs are failing when trying to retrieve v2alpha1 configurations with only v2 client...

Reply:
Right, I'm saying to just use the v2alpha1 client even in v1.16.

Reply:
+1 to Casey's suggestion. It will simplify the code. It's good that this is possible as that's what I was trying to get at in https://github.com/cilium/cilium/pull/31721#discussion_r1548403067.